### PR TITLE
Server routine: is_relevant functions

### PR DESCRIPTION
**Pull request by jparisu**
New branch to merge ddb_functions
is_relevant
update
force c++14
merge and compile everything

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2703/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/94/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2500/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2714/)

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 1]
```diff
@@ -33,27 +33,69 @@ bool DiscoveryDataBase::pdp_is_relevant(
...
-    (void)reader_guid;
-    return true;
+    auto itp = participants_.find(guid_from_change(&change).guidPrefix);
+    if (itp == participants_.end())
+    {
+        // not relevant
+        return false;
+    }
+
+    if (!itp->second.is_matched(reader_guid.guidPrefix))
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:58]:
```suggestion
    }
    else if (!itp->second.is_matched(reader_guid.guidPrefix))
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 2]
```diff
@@ -33,27 +33,69 @@ bool DiscoveryDataBase::pdp_is_relevant(
...
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:85]:
```suggestion
    }
    else if (!itp->second.is_matched(reader_guid.guidPrefix))
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 3]
```diff
@@ -33,27 +33,69 @@ bool DiscoveryDataBase::pdp_is_relevant(
         const eprosima::fastrtps::rtps::CacheChange_t& change,
         const eprosima::fastrtps::rtps::GUID_t& reader_guid) const
 {
-    (void)change;
-    (void)reader_guid;
-    return true;
+    auto it = participants_.find(guid_from_change(&change).guidPrefix);
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:36]: In general, don't we need to lock the mutex before any access to the data structures?

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 4]
```diff
@@ -69,10 +111,12 @@ void DiscoveryDataBase::add_ack_(
...
         eprosima::fastrtps::rtps::CacheChange_t* change,
-        std::string topic_name)
+        eprosima::fastrtps::string_255 topic_name)
 {
-    (void)change;
-    (void)topic_name;
+    //  add the data to the queue to process
+    data_queue_.Push(eprosima::fastdds::rtps::ddb::DiscoveryDataQueueInfo(change, topic_name));
+
+    // not way to check if is an error
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:119]: I don't understand this error propagation

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 5]
```diff
@@ -146,17 +190,16 @@ bool DiscoveryDataBase::process_data_queue()
...
     // Process all messages in the queque
     while (!data_queue_.Empty())
     {
         // Process each message with Front()
-        data_queue_info = data_queue_.Front();
-        change = data_queue_info.change();
-        topic_name = data_queue_info.topic();
+        DiscoveryDataQueueInfo data_queue_info = data_queue_.Front();
+        change = data_queue_info.cache_change();
+        topic_name = data_queue_info.topic_name();
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:202]: Here seems that you haven't rebase the data_queue branch.

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 6]
```diff
@@ -680,29 +723,27 @@ bool DiscoveryDataBase::process_dirty_topics()
...
         fastrtps::rtps::CacheChange_t* change)
 {
-    (void)change;
-    return true;
-    /*
-       if (change->kind != fastrtps::rtps::ChangeKind_t::ALIVE)
-       {
-        logWarning(DISCOVERY_DATABASE, "Attempting to delete information of an ALIVE entity: " << guid_from_change_(change));
+
+    if (change->kind != fastrtps::rtps::ChangeKind_t::ALIVE)
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:727]: According to the logWarning it should be like this:

```suggestion
    if (change->kind == fastrtps::rtps::ChangeKind_t::ALIVE)
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp [hunk 7]
```diff
@@ -31,27 +31,33 @@ class DiscoveryDataQueueInfo
...
     }
 
     ~DiscoveryDataQueueInfo()
     {
     }
 
-    eprosima::fastrtps::rtps::CacheChange_t* change();
-
-    eprosima::fastrtps::string_255 topic();
+    eprosima::fastrtps::rtps::CacheChange_t* cache_change()
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp:46]: In line with the rest of the structures and the design:
```suggestion
    eprosima::fastrtps::rtps::CacheChange_t* change()
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp [hunk 8]
```diff
@@ -31,27 +31,33 @@ class DiscoveryDataQueueInfo
...
+    {
+        return cache_change_;
+    }
 
-    eprosima::fastrtps::rtps::GUID_t associated_entity();
+    eprosima::fastrtps::string_255 topic_name()
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp:51]:
```suggestion
    eprosima::fastrtps::string_255 topic()
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp [hunk 9]
```diff
@@ -31,27 +31,33 @@ class DiscoveryDataQueueInfo
...
+    {
+        return topic_name_;
+    }
 
 private:
 
-    eprosima::fastrtps::rtps::CacheChange_t* change_;
-
-    eprosima::fastrtps::string_255 topic_;
+    eprosima::fastrtps::rtps::CacheChange_t* cache_change_;
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp:58]:
```suggestion
    eprosima::fastrtps::rtps::CacheChange_t* change_;
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp [hunk 10]
```diff
@@ -31,27 +31,33 @@ class DiscoveryDataQueueInfo
...
 
-    eprosima::fastrtps::rtps::GUID_t associated_entity_;
+    eprosima::fastrtps::string_255 topic_name_;
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.hpp:60]:
```suggestion
    eprosima::fastrtps::string_255 topic_;
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2707/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/96/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2504/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2718/)

**Pull request merged**